packages/core/src/bus/BusInstruction.ts

```diff
@@ -0,0 +1,27 @@
+import { randomBytes } from 'crypto';
+
+export abstract class BusInstruction<T> {
+  public readonly type!: string;
+  public readonly payload!: T;
+  public readonly correlationId!: string;
+  public readonly createdAt!: Date;
+
+  constructor(
```

Comment:
The abstract class must have the protected constructor
```suggestion
  protected constructor(
```